docx_editor/track_changes.py

```
@@ -214,6 +224,87 @@ def _apply_single_edit(self, op: EditOperation) -> int:
+            if not op.anchor or op.text is None:
+                return f"{op.action} requires 'anchor' and 'text'"
+            target = op.anchor
+        else:
+            return f"Unknown action: {op.action}"
+
+        # The find never raises for well-formed input, but a dry-run must not
+        # throw for any input (e.g. a negative ``occurrence``); report instead.
+        try:
+            if self._find_in_paragraph(p, target, op.occurrence) is None:
+                return f"text {target!r} not found in paragraph {op.paragraph} ({self._paragraph_preview(p)!r})"
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)